genkit-tools/cli/src/utils/updates.ts

```diff
@@ -0,0 +1,246 @@
+}
+
+/**
+ * Returns the current CLI version, normalized.
+ */
+export function getCurrentVersion(): string {
+  return normalizeVersion(currentVersion);
+}
+
+/**
+ * Normalizes a version string by removing a leading 'v' if present.
+ * @param version - The version string to normalize
+ * @returns The normalized version string
+ */
+function normalizeVersion(version: string): string {
+  return version.replace(/^v/, '');
+}
+
+/**
+ * Interface for the Google Cloud Storage latest.json response
+ */
+interface GCSLatestResponse {
+  channel: string;
+  latestVersion: string;
+  lastUpdated: string;
+  platforms: Record<
+    string,
+    {
+      url: string;
+      version: string;
+      versionedUrl: string;
+    }
+  >;
+}
+
+/**
+ * Interface for npm registry response
+ */
+interface NpmRegistryResponse {
+  'dist-tags': {
+    latest: string;
+    [key: string]: string;
+  };
+  versions: Record<string, any>;
+}
+
+/**
+ * Fetches the latest release data from GCS.
+ */
+async function getGCSLatestData(): Promise<GCSLatestResponse> {
+  const response = await AXIOS_INSTANCE.get(`${GCS_BUCKET_URL}/latest.json`);
+
+  if (response.status !== 200) {
+    throw new GenkitToolsError(
+      `Failed to fetch GCS latest.json: ${response.statusText}`
+    );
+  }
+
+  return response.data as GCSLatestResponse;
+}
+
+/**
+ * Gets the latest CLI version from npm registry for non-binary installations.
+ * @param ignoreRC - If true, ignore prerelease versions (default: true)
+ */
+export async function getLatestVersionFromNpm(
+  ignoreRC: boolean = true
+): Promise<string | null> {
+  try {
+    const response = await AXIOS_INSTANCE.get(
+      `https://registry.npmjs.org/${name}`
+    );
+
+    if (response.status !== 200) {
+      throw new GenkitToolsError(
+        `Failed to fetch npm versions: ${response.statusText}`
+      );
+    }
+
+    const data: NpmRegistryResponse = response.data;
+
+    // Prefer dist-tags.latest if valid and not a prerelease (if ignoreRC)
+    const latest = data['dist-tags']?.latest;
+    if (latest) {
+      const clean = normalizeVersion(latest);
+      if (semver.valid(clean) && (!ignoreRC || !semver.prerelease(clean))) {
+        return clean;
+      }
+    }
+
+    // Fallback: find the highest valid version in versions
+    const versions = Object.keys(data.versions)
+      .map(normalizeVersion)
+      .filter((v) => semver.valid(v) && (!ignoreRC || !semver.prerelease(v)));
+
+    if (versions.length === 0) {
+      return null;
+    }
+
+    // Sort by semver descending (newest first)
+    versions.sort(semver.rcompare);
+    return versions[0];
+  } catch (error: any) {
+    if (error instanceof GenkitToolsError) {
+      throw error;
+    }
+
+    throw new GenkitToolsError(
+      `Failed to fetch npm versions: ${error?.message ?? error}`
+    );
+  }
+}
+
+/**
+ * Checks if update notifications are disabled via environment variable or user config.
+ */
+function areUpdateNotificationsDisabled(): boolean {
```

Comment:
I know it's not grammatically correct, but let's stick with `is*`... 

```suggestion
function isUpdateNotificationsDisabled(): boolean {
```